File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetDeviceGroupSurfacePresentModesKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetDeviceGroupSurfacePresentModesKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong surface, jlong pModes, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetDeviceGroupSurfacePresentModesKHR f = (PFN_vkGetDeviceGroupSurfacePresentModesKHR)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkSurfaceKHR*)jniLongToPointer(surface)),
        ((VkDeviceGroupPresentModeFlagsKHR*)jniLongToPointer(pModes))
    );
}


jint Java_cz_mg_vulkan_PFNvkGetDeviceGroupSurfacePresentModesKHR_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong surface, jlong pModes){
    (void)env;
    (void)jc;
    PFN_vkGetDeviceGroupSurfacePresentModesKHR f = (PFN_vkGetDeviceGroupSurfacePresentModesKHR)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkSurfaceKHR*)jniLongToPointer(surface)),
        ((VkDeviceGroupPresentModeFlagsKHR*)jniLongToPointer(pModes))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkGetDeviceGroupSurfacePresentModesKHR.java
package cz.mg.vulkan;

public class PFNvkGetDeviceGroupSurfacePresentModesKHR extends VkFunctionPointer {
    public PFNvkGetDeviceGroupSurfacePresentModesKHR() {
    }

    protected PFNvkGetDeviceGroupSurfacePresentModesKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetDeviceGroupSurfacePresentModesKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetDeviceGroupSurfacePresentModesKHR(long value) {
        setValue(value);
    }

    public PFNvkGetDeviceGroupSurfacePresentModesKHR(VkInstance instance) {
        super(instance, new VkString("vkGetDeviceGroupSurfacePresentModesKHR"));
    }

    public void call(VkDevice device, VkSurfaceKHR surface, VkDeviceGroupPresentModeFlagsKHR pModes, VkResult rval){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), surface != null ? surface.getVkAddress() : VkPointer.getNullAddressNative(), pModes != null ? pModes.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long device, long surface, long pModes, long rval);


    public int call(VkDevice device, VkSurfaceKHR surface, VkDeviceGroupPresentModeFlagsKHR pModes){
        return callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), surface != null ? surface.getVkAddress() : VkPointer.getNullAddressNative(), pModes != null ? pModes.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callSimplifiedNative(long vkaddress, long device, long surface, long pModes);

}
